### PR TITLE
Update release script

### PR DESCRIPTION
**Pull request by hcoggins**
An option for --dry-run was added, and we now release from a temporary branch, since we cannot push onto main.

### CODE REVIEW

File: release.sh
Comment (NiklasEi) [inline release.sh:39]: Are we sure at this point that we are branching from an up to date main?

File: release.sh
Comment (NiklasEi) [inline release.sh:88]: Don't we want to keep the remote release branch and merge it? This looks to me as if we would not have our version tag on main at the moment after a release.

Comment (NiklasEi) [inline release.sh:88]: We are using `origin` as the remote name for the central repository. But e.g. for me this would not work (`origin` points to my own fork of momo while `upstream` points to `TNG/momo-scheduler`)
We could try coming up with something nicer here (can we identify the remote based on the URL?), or need to document that the remote `origin` needs to point to `TNG/momo-scheduler`.

Comment (weissu42) [reply release.sh:88]: Yes, we should merge the release branch to main after we committed the updated package.json
I guess we need to wait for the github checks to succeed or our github rules won't allow us to merge ... so somehow sleep a while, try to merge, if checks are not ready, sleep more, try again ...?

File: release.sh
Comment (weissu42) [reply release.sh:39]: maybe we can just add a 'git pull' - if that fails, because you have uncommitted files lying around, that's totally what we want, isn't it?

Comment (hcoggins) [reply release.sh:39]: good point - we can add this

File: release.sh
Comment (hcoggins) [reply release.sh:88]: Ah ok, I wasn't aware we wanted to merge this back after the release. Is it possible to merge onto main without having a pull request and review, since we have _main_ as a protected branch?

Comment (hcoggins) [reply release.sh:88]: Could we just used the url in the command directly?

Comment (weissu42) [reply release.sh]: good question, I don't now, maybe we will need to remove the requirement for reviews to be able to do this ... :(

File: release.sh
Comment (weissu42) [inline release.sh:8]: suggestion: set -eux
u treats unset variables as error
x prints every command that is executed

File: release.sh
Comment (weissu42) [inline release.sh:35]: maybe also use the variable here? echo "You do not seem to be on the ${UPSTREAM_BRANCH} branch!"

File: release.sh
Comment (weissu42) [inline release.sh:65]: our echos have different styles, some ending in ":", some in "...", some in nothing, can we unify that?

File: release.sh
Comment (NiklasEi) [reply release.sh]: We decided to split the release script in two parts. The first script will prepare everything and create a PR. After manually merging the PR the second script will do the actual publishing to npmjs.

**Review by weissu42 — APPROVED** [review]

**Pull request merged**
